internal/service/organizations/organizational_unit_descendant_accounts_test.go

```diff
@@ -0,0 +1,73 @@
+package organizations_test
+
+import (
+	"testing"
+
+	"github.com/aws/aws-sdk-go/service/organizations"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-provider-aws/internal/acctest"
+)
+
+func testAccOrganizationalUnitDescendantAccountsDataSource_basic(t *testing.T) {
```

Comment:
Not sure if the Terraform source tests allow for this behavior, but I assume you want to check all them. If possible, you can try the `t.Run` method with `0, 1, 2, 3` for that `resource.TestCheckResourceAttr(dataSourceName, "accounts.#", "3"),` line. https://go.dev/blog/subtests